redis/runtime/src/main/java/io/quarkiverse/langchain4j/redis/runtime/RedisEmbeddingStoreRecorder.java

```diff
@@ -0,0 +1,43 @@
+package io.quarkiverse.langchain4j.redis.runtime;
+
+import java.util.Collections;
+import java.util.function.Supplier;
+
+import jakarta.enterprise.inject.Default;
+
+import io.quarkiverse.langchain4j.redis.RedisEmbeddingStore;
+import io.quarkus.arc.Arc;
+import io.quarkus.redis.client.RedisClientName;
+import io.quarkus.redis.datasource.ReactiveRedisDataSource;
+import io.quarkus.redis.runtime.client.config.RedisConfig;
+import io.quarkus.runtime.annotations.Recorder;
+
+@Recorder
+public class RedisEmbeddingStoreRecorder {
+
+    public Supplier<RedisEmbeddingStore> embeddingStoreSupplier(RedisEmbeddingStoreConfig config) {
+        RedisEmbeddingStore.Builder builder = new RedisEmbeddingStore.Builder();
+
+        String redisClientName = config.redisClientName();
+        if (redisClientName.equals(RedisConfig.DEFAULT_CLIENT_NAME)) {
+            builder.dataSource(Arc.container().select(ReactiveRedisDataSource.class, new Default.Literal()).get());
+        } else {
+            builder.dataSource(
+                    Arc.container().select(ReactiveRedisDataSource.class, new RedisClientName.Literal(redisClientName)).get());
+        }
```

Comment:
This is risky in synthetic beans. Using `SyntheticCreationalContext`

Reply:
See what we do for Flyway: https://github.com/quarkusio/quarkus/blob/7e23cf507d30cbd32e5aa65102d0fc519ab6264f/extensions/flyway/deployment/src/main/java/io/quarkus/flyway/deployment/FlywayProcessor.java#L191

Reply:
The general idea here is that when producing a synthetic bean, one should absolutely declare the injection points instead of trying to pull the bean from under the hood which can fail very easily due to timing issues

Reply:
Yeah, I got bitten by this just yesterday. 
You need to predeclare your injection point in the synthetic bean and use the creation context to retrieve it. I can send you some examples if you want (unfortunately, the code I wrote yesterday is not yet open).